### PR TITLE
Feat/df 52 get cash flows

### PR DESCRIPTION
**Pull request by shawndcarpenter**
### Keypoints:
[what changes have been made]
- Adds cashFlow type
- Uses SQL query to create cashFlows
- Adds testing
- Changes dates on income and expense factories to be more dynamic
- Changes source to vendor and type to status for transactions
### Related Issue(s)
closes DF-52

### Checklist:
- [x] New/Modified Tests are Passing
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings or errors.

#### Additional Notes:
[Please include specific lines/files affected if needed]
[Any special testing/config considerations]

### CODE REVIEW

File: app/graphql/types/month_type.rb
Comment (trpubz) [inline app/graphql/types/month_type.rb:1]: I see this a custom type that lives in the YearType, but I don't see a spec testing the output

File: app/graphql/types/year_type.rb
Comment (trpubz) [inline app/graphql/types/year_type.rb:1]: if these were from iteration 1, then let's delete the YearType and MonthType files

File: app/graphql/types/cash_flow_type.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module Types
+  class CashFlowType < Types::BaseObject
+    # field :username, String, null: true
+    # field :years, [Types::YearType], null: true
```
Comment (trpubz) [inline app/graphql/types/cash_flow_type.rb:6]: I now see the YearType custom, I recommend deleting this reference and other 2 files;
we can always reference the previous commit if we want to explore, but as for data going into main, should be removed

**Review by trpubz — COMMENTED** [review]
This looks good, please see my 3 comments, no action may be needed

**Review by josephlee702 — APPROVED** [review]
Just double checked everything and we seem to be good here. Output is coming out as expected. Merging in.

**Pull request merged**